thunder/tests/distributed/test_dtensor.py

```diff
@@ -0,0 +1,96 @@
+import unittest
+
+import pytest
+import torch
+
+if not torch.distributed.is_available():
+    pytest.skip(allow_module_level=True)
+
+from thunder.dynamo import thunderfx
+import thunder
+
+from thunder.tests.distributed.helper import DistributedParallelTestCase
+from torch.distributed._tensor import DeviceMesh, Shard, distribute_tensor
+from torch.distributed.tensor.placement_types import Placement, Shard, Replicate
+
+
+@unittest.skipUnless(
+    torch.cuda.is_available() and torch.distributed.is_available() and torch.distributed.is_nccl_available(),
```

Comment:
nit-picking:
```suggestion
    torch.cuda.is_available() and torch.distributed.is_nccl_available(),
```

We already check the `torch.distributed` availability at line 6.